### PR TITLE
Minor improvement on error messages

### PR DESCRIPTION
**Pull request by PooyaKharamesh**
General improvement on error messages docs and ...

### CODE REVIEW

File: Actions/PipelineCleanup/PipelineCleanup.ps1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Param(
     [Parameter(HelpMessage = "Project folder", Mandatory = $false)]
     [string] $project = ".",
-    [Parameter(HelpMessage = "Specifies the parent telemetry scope for the Telemetry signal", Mandatory = $false)]
+    [Parameter(HelpMessage = "Specifies the parent telemetry scope for the telemetry signal", Mandatory = $true)]
```
Comment (freddydk) [inline Actions/PipelineCleanup/PipelineCleanup.ps1:4]: If you set it to true here, you also need to set it to true in the action.yaml file

File: Actions/CheckForUpdates/CheckForUpdates.ps1 [hunk 2]
```diff
@@ -140,18 +140,13 @@ try {
...
 
     if (-not $update) {
         if (($updateFiles) -or ($removeFiles)) {
-            OutputWarning -message "Updated AL-Go System Files are available for your repository, please run the Update AL-Go System Files workflow"
+            OutputWarning -message "There are updates for your AL-Go system, run 'Update AL-Go System Files' workflow to download the latest version of AL-Go."
         }
         else {
-            Write-Host "No updated AL-Go System Files are available"
+            Write-Host "Your repository runs on the latest version of AL-Go System."
         }
```
Comment (freddydk) [inline Actions/CheckForUpdates/CheckForUpdates.ps1:150]: Technically it is wrong to call it the latest version of AL-Go System
A repo that was created using v0.1 might be up-to-date with v0.1, but a v0.2 exists.
This means that he would have the latest files in v0.1 but a newer version is available (not yet detected by the script)

File: Actions/AddExistingApp/action.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ inputs:
     required: false
     default: ${{ github.token }}
   parentTelemetryScopeJson:
-    description: Specifies the parent telemetry scope for the Telemetry signal
+    description: Specifies the parent telemetry scope for the telemetry signal
     required: false
```
Comment (freddydk) [inline Actions/AddExistingApp/action.yaml:17]: this needs to be true if you change the .ps1 required to be true - and then the default should be removed.

File: Actions/AddExistingApp/AddExistingApp.ps1 [hunk 4]
```diff
@@ -3,7 +3,7 @@ Param(
     [string] $actor,
     [Parameter(HelpMessage = "The GitHub token running the action", Mandatory = $false)]
     [string] $token,
-    [Parameter(HelpMessage = "Specifies the parent telemetry scope for the Telemetry signal", Mandatory = $false)]
+    [Parameter(HelpMessage = "Specifies the parent telemetry scope for the telemetry signal", Mandatory = $true)]
     [string] $parentTelemetryScopeJson = '{}',
```
Comment (freddydk) [inline Actions/AddExistingApp/AddExistingApp.ps1:7]: if it is mandatory - there shouldn't be a default.
I think you should leave those as mandatory false.

**Comment by freddydk** [comment]
Unit tests are failing - but the job succeeds???
image

File: Actions/PipelineCleanup/PipelineCleanup.ps1 (hunk 1 above)
Comment (PooyaKharamesh) [reply Actions/PipelineCleanup/PipelineCleanup.ps1]: thinking again about it I dont believe that it should be mandatory. Ill fix it accordingly

File: Actions/CheckForUpdates/CheckForUpdates.ps1 (hunk 2 above)
Comment (PooyaKharamesh) [reply Actions/CheckForUpdates/CheckForUpdates.ps1:150]: I also thought about it. but if scripts is not detecting any new update it means that they run on the latest. IMO we should update the script to detect a new branch and report it?

File: Actions/AddExistingApp/action.yaml (hunk 3 above)
Comment (PooyaKharamesh) [reply Actions/AddExistingApp/action.yaml:17]: Rethinking about it I believe it shouldnt be mandatory so Ill fix it accordingly

**Pull request merged**